chromadb/db/impl/sqlite.py

```diff
@@ -246,3 +257,39 @@ def uuid_to_db(uuid: Optional[UUID]) -> Optional[Any]:
     @override
     def unique_constraint_error() -> Type[BaseException]:
         return sqlite3.IntegrityError
+
+    @trace_method("SqliteDB.optimize_system", OpenTelemetryGranularity.ALL)
+    @override
+    def optimize_system(self) -> SystemOptimizationStats:
+        super().optimize_system()
+        # get size of DB file
+        if not self._is_persistent:
+            # we don't need to optimize an in-memory DB
+            return SystemOptimizationStats(
+                db_size_before=-1,
+                db_size_after=-1,
+            )
+        file_size_before = os.path.getsize(self._db_file)
+        self._conn_pool.close()
```

Comment:
@HammadB, we do this also in `reset(),` but I think in multi-user env where ongoing transactions may happen, there is an inherent risk of losing data or corrupting the DB.  One way to do this is to use check-in/checkout for connections and wait until all connections have checked in.